sympy/matrices/expressions/matpow.py

```diff
@@ -56,14 +56,25 @@ def doit(self, **kwargs):
             args = self.args
         base = args[0]
         exp = args[1]
-        if isinstance(base, MatrixBase) and exp.is_number:
+        if exp.is_zero and base.is_square:
+            try:
```

Comment:
This was done to deal with special cases, such as instances of Immutable matrix (being raised to power 0), where I have to return it in the form of base.__class__(Identity(base.shape[0])), and not just as Identity((base.shape[0])).

If not done so, there will be issues while dealing with Immutable matrix instance or other similar ones. (It did lead to assertion errors in test cases in previous commits)